lib/elixir/test/elixir/enum_test.exs

```diff
@@ -417,6 +417,59 @@ defmodule EnumTest do
     assert Enum.into(["H", "i"], "") == "Hi"
   end
 
+  test "into/2 exceptions" do
+    item = 1
+    date = ~D[2015-01-01]
+    map = %{a: 1}
+    tuple = {:a, 1}
+
+    map_msg = "collecting into a map requires {key, value} tuples, got: #{inspect(item)}"
+    map_date_msg = "collecting into a map requires {key, value} tuples, got: #{inspect(date)}"
+    assert_raise ArgumentError, map_msg, fn -> Enum.into(Range.new(item, 10, 1), %{}) end
+
+    assert_raise ArgumentError, map_date_msg, fn ->
+      Enum.into(Date.range(date, Date.new!(2015, 1, 3)), %{})
```

Comment:
We typically use sigil to represent date litterals
```suggestion
      Enum.into(Date.range(date, ~D[2015-01-03]), %{})
```

Reply:
Actually, for the purpose of this test, I think testing `Date.range` might be overkill.
We're not doing it for `test "into/2" do` above, and it checks the same logic anyway.